addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -290,13 +290,7 @@ def _search_virtual_remaining_leaves(self, operator, value):
     @api.depends_context('employee_id', 'default_employee_id', 'default_date_from')
     def _compute_leaves(self):
         employee = self.env['hr.employee']._get_contextual_employee()
-        target_date = self._context['default_date_from'] if 'default_date_from' in self._context else None
-        # This is a workaround to save the date value in context for next triggers
-        # when context gets cleaned and 'default_' context keys gets removed
-        if target_date:
-            self.env.context = frozendict(self.env.context, leave_date_from=self._context['default_date_from'])
-        else:
-            target_date = self._context.get('leave_date_from', None)
+        target_date = self.env.context.get('default_date_from')
```

Comment:
Won't this behavior be effectively modified if this is dropped?

Reply:
Yes, I'm trying to solve this workaround/hack. But no luck yet: https://github.com/odoo/odoo/pull/188584#discussion_r1968031862

If you have any ideas on how to solve it properly, I am all ears.